File: SDKs/Aspose_Cloud_SDK_For_Ruby/lib/imaging/converter.rb
module Aspose
  module Cloud
    module Imaging
      class Converter
        def initialize(filename)
          @filename = filename
          raise 'filename not specified.' if filename.empty?
          @base_uri =  Aspose::Cloud::Common::Product.product_uri + '/imaging/' + @filename
        end

=begin
  convert an image file to a different format
  @param string input_file_path Path of the input file.
  @param string output_filename Name of the output file.
  @param string save_format Output file format.
=end
        def convert_local_file(input_file_path, output_filename, save_format)
          raise 'input_file_path not specified.' if input_file_path.empty?
          raise 'output_filename not specified.' if output_filename.empty?
          raise 'save_format not specified.' if save_format.empty?

          str_uri = "#{Aspose::Cloud::Common::Product.product_uri}/imaging/#{@filename}/saveAs?format=#{save_format}"
          signed_str_uri = Aspose::Cloud::Common::Utils.sign(str_uri)
          response_stream = RestClient.get(signed_str_uri, {:accept=>'application/json', :payload=>File.new(input_file_path, 'rb')})

          valid_output = Aspose::Cloud::Common::Utils.validate_output(response_stream)

          if valid_output.empty?
            save_format = 'zip' if save_format.eql?('html')
            output_path = "#{Aspose::Cloud::Common::AsposeApp.output_location}#{Aspose::Cloud::Common::Utils.get_filename(output_filename)}.#{save_format}"
            Aspose::Cloud::Common::Utils.save_file(response_stream,output_path)
          end
          valid_output
        end

=begin
  convert an image file to a fax
=end
        def convert_tiff_to_fax(folder_name = '', storage_type = 'Aspose', storage_name = '')
          str_uri = "#{Aspose::Cloud::Common::Product.product_uri}/imaging/tiff/#{@filename}/toFax"
          signed_str_uri = Aspose::Cloud::Common::Utils.sign(str_uri)
          response_stream = RestClient.get(signed_str_uri, {:accept=>'application/json'})

          valid_output = Aspose::Cloud::Common::Utils.validate_output(response_stream)

          if valid_output.empty?
            output_path = "#{Aspose::Cloud::Common::AsposeApp.output_location}#{@filename}"
            Aspose::Cloud::Common::Utils.save_file(response_stream,output_path)
          end
          valid_output
        end
      end
    end
  end
end

File: Plugins/Aspose_Cloud_for_WordPress/aspose-cloud-presentation-importer/Aspose_Cloud_SDK_For_PHP-master/src/Aspose/Cloud/Slides/Converter.php
<?php
/**
 * Converts pages or document into different formats.
 */
namespace Aspose\Cloud\Slides;

use Aspose\Cloud\Common\AsposeApp;
use Aspose\Cloud\Common\Product;
use Aspose\Cloud\Common\Utils;
use Aspose\Cloud\Exception\AsposeCloudException as Exception;

class Converter
{

    public $fileName = '';
    public $saveFormat = '';

    public function __construct($fileName, $saveFormat = 'PPT')
    {
        //set default values
        $this->fileName = $fileName;

        $this->saveFormat = $saveFormat;
    }

    /**
     * Saves a particular slide into various formats with specified width and height.
     *
     * @param integer $slideNumber The number of slide.
     * @param string $imageFormat The image format.
     *
     * @return string Returns the file path.
     * @throws Exception
     */
    public function convertToImage($slideNumber, $imageFormat)
    {


        $strURI = Product::$baseProductUri . '/slides/' . $this->getFileName() . '/slides/' . $slideNumber . '?format=' . $imageFormat;

        $signedURI = Utils::sign($strURI);

        $responseStream = Utils::processCommand($signedURI, 'GET', '', '');
        $v_output = Utils::validateOutput($responseStream);
        if ($v_output == '') {
            $outputPath = AsposeApp::$outPutLocation . Utils::getFileName($this->getFileName()) . '.' . $imageFormat;
            Utils::saveFile($responseStream, $outputPath);
            return $outputPath;
        } else {
            return $v_output;
        }
    }

    /**
     * Convert a particular slide into various formats with specified width and height.
     *
     * @param integer $slideNumber The slide number.
     * @param string $imageFormat The image format.
     * @param integer $width The width of image.
     * @param integer $height The height of image.
     *
     * @return string Returns the file path.
     * @throws Exception
     */
    public function convertToImagebySize($slideNumber, $imageFormat, $width, $height)
    {


        $strURI = Product::$baseProductUri . '/slides/' . $this->getFileName() . '/slides/' . $slideNumber . '?format=' . $imageFormat . '&width=' . $width . '&height=' . $height;

        $signedURI = Utils::sign($strURI);

        $responseStream = Utils::processCommand($signedURI, 'GET', '', '');
        $v_output = Utils::validateOutput($responseStream);
        if ($v_output == '') {
            $outputPath = AsposeApp::$outPutLocation . 'output.' . $imageFormat;
            Utils::saveFile($responseStream, $outputPath);
            return $outputPath;
        } else {
            return $v_output;
        }
    }

    /**
     * Convert a document to the specified format.
     *
     * @return string Returns the file path.
     * @throws Exception
     */
    public function convert()
    {


        $strURI = Product::$baseProductUri . '/slides/' . $this->getFileName() . '?format=' . $this->saveFormat;

        $signedURI = Utils::sign($strURI);
        $responseStream = Utils::processCommand($signedURI, 'GET', '', '');

        $v_output = Utils::validateOutput($responseStream);

        if ($v_output === '') {
            return $responseStream;
        } else {
            return $v_output;
        }
    }
    
    /**
     * Convert a document to SaveFormat without using Aspose storage.
     * 
     * @param string $inputPath The path of source file.
     * @param string $outputFile Path where you want to file after conversion.
     * @param string $saveFormat New file format.
     * 
     * @return string|boolean Return the file path.
     * @throws Exception
     */
    public function convertLocalFile($inputPath, $outputFile, $saveFormat) {
        if ($inputPath == '')
            throw new Exception('Please specify input file');
            
        if ($outputFile == '') 
            throw new Exception('Please specify output file');
            
        if ($saveFormat == '') 
            throw new Exception('Please specify save format');
        
        $strURI = Product::$baseProductUri . '/slides/convert?format=' . $saveFormat;

        $signedURI = Utils::sign($strURI);
        
        $responseStream = Utils::uploadFileBinary($signedURI, $inputPath, 'xml');        

        $v_output = Utils::validateOutput($responseStream);
        
        if ($v_output === '') {
            if ($saveFormat == 'html') {
                $outputFormat = 'zip';
            } else {
                $outputFormat = $saveFormat;
            }
            
            $outputFileName = Utils::getFileName($outputFile) . '.' . $outputFormat;
            Utils::saveFile($responseStream, AsposeApp::$outPutLocation . $outputFileName);
            return $outputFileName;
        } else {
            return $v_output;
        }
    }

    /**
     * @return string
     */
    public function getFileName()
    {
        if ($this->fileName == '') {
            throw new Exception('No File Name Specified');
        }
        return $this->fileName;
    }

    /**
     * @param string $fileName
     */
    public function setFileName($fileName)
    {
        $this->fileName = $fileName;
        return $this;
    }

    /**
     * @return string
     */
    public function getSaveFormat()
    {
        return $this->saveFormat;
    }

    /**
     * @param string $saveFormat
     */
    public function setSaveFormat($saveFormat)
    {
        $this->saveFormat = $saveFormat;
        return $this;
    }

}